### PR TITLE
Upgraded `test-runner` image to latest.

### PR DESCRIPTION
**Pull request by bendory**
# Changes

This change will update our plumbing to include
https://github.com/tektoncd/plumbing/pull/1308
as well as any other changes since the last upgrade.

Context: https://github.com/tektoncd/pipeline/pull/5918#issuecomment-1366766687
Issue: https://github.com/tektoncd/pipeline/issues/5899

/kind cleanup

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [N/A] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -1165,7 +1165,7 @@ presubmits:
     trigger: "(?m)^/test (all|pull-tekton-pipeline-build-tests),?(\\s+|$)"
     spec:
       containers:
-      - image: gcr.io/tekton-releases/dogfooding/test-runner@sha256:01bace96516d22d70c092ff9c01f9ffe7af806c7475157dcaa7cdb45d84bd68b # golang 1.18.7
```
Comment (abayer) [inline prow/config.yaml:1168]: It would probably be worth replacing this SHA in all the jobs.

**Review by abayer — CHANGES_REQUESTED** [review]

File: prow/config.yaml (hunk 1 above)
Comment (bendory) [reply prow/config.yaml:1168]: Doh! I naively thought this would be defined in one place and used everywhere.

**Pull request merged**